hs_core/page_processors.py

```diff
@@ -34,6 +35,16 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
 
     metadata_status = _get_metadata_status(content_model)
 
+    relevant_tools = []
+    for tool in ToolResource.objects.all():
+        if len(tool.metadata.res_types.all())>0:
```

Comment:
but really, if you are only checking the length to see if you should loop over them, just do the loop directly. the conditional isn't accomplishing anything here.

In fact, you could probably make this one query instead of 1+N queries, but its hard to say with the new models apparently missing from this PR.